packages/interchain/interchain-core/src/packet.rs

```diff
@@ -0,0 +1,335 @@
+use cosmwasm_std::{Binary, StdError};
+use cw_orch_core::environment::CwEnv;
+use cw_orch_core::environment::IndexResponse;
+use ibc_relayer_types::core::{
+    ics04_channel::packet::Sequence,
+    ics24_host::identifier::{ChannelId, PortId},
+};
+
+use crate::{results::NetworkId, tx::TxId};
+
+/// Structure to hold simple information about a sent packet
+#[derive(Debug, Clone)]
+pub struct IbcPacketInfo {
+    /// Port on which is packet was sent
+    pub src_port: PortId,
+    /// Channel on which is packet was sent
+    pub src_channel: ChannelId,
+    /// Packet identification (sequence is `u64` number)
+    pub sequence: Sequence,
+    /// Chain identification to which the packet was sent
+    pub dst_chain_id: NetworkId,
+}
+
+/// Raw packet outcome
+/// The T generic is used to allow for raw transactions or analyzed transactions to be used
+#[derive(Debug, PartialEq, Clone)]
+#[must_use = "We recommend using `PacketAnalysis::into_result()` to assert ibc success"]
+pub enum IbcPacketOutcome<T> {
```

Comment:
What's the difference between this type and `SuccessIbcPacketOutcome`? 

It seems like the `Success` variants should be `SuccessIbcPacketOutcome`?

Reply:
IbcPacketOutcome is the raw outcome from the packet cycle awaiting. It tells you what the packet lifecycle looks like for this packet. 
The Success variant corresponds to lifecycle with an ack. But not to a successful ack, because nothing has been parsed yet

Reply:
I would like us to differentiate between the low-level IBC lifecycle and whatever lifecycle the application expects. 

A packet either has a timeout or is acknowledged. Having an acknowledged packet doesn't give any guarantees about the data with that ack, and that is OK but we should be very precise with our names. 

I propose we use the IBC-native naming conventions to talk about the packet lifecycle (send, commit, ack, timeout) and we don't reuse those terms when talking about the other assertions that could be done on the data within the ack. 

We can then also get rid of terms like "success". If the packet is acknowledged then that's enough to proceed to the next step in the flow.